pandas/tseries/offsets.py

```diff
@@ -381,6 +381,8 @@ def __call__(self, other):
     def __add__(self, other):
         if isinstance(other, (ABCDatetimeIndex, ABCSeries)):
             return other + self
+        elif isinstance(other, ABCPeriod):
```

Comment:
I don't think these are intended ATM... Should cover (part of) #12759 in this PR?

```
pd.Period('2011-01', freq='M') + pd.NaT
# NaT

pd.NaT + pd.Period('2011-01', freq='M')
# TypeError: unsupported operand type(s) for +: 'NaTType' and 'pandas._period.Period'
```